internal/controller/ols_app_server_reconciliator.go

```diff
@@ -236,3 +243,38 @@ func (r *OLSConfigReconciler) reconcileService(ctx context.Context, cr *olsv1alp
 	r.logger.Info("OLS service reconciled", "service", service.Name)
 	return nil
 }
+
+func (r *OLSConfigReconciler) reconcileLLMSecrets(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	providerCredentials := ""
+	var providerApiToken string
+	for _, provider := range cr.Spec.LLMConfig.Providers {
+		foundSecret := &corev1.Secret{}
+		err := r.Client.Get(ctx, client.ObjectKey{Name: provider.CredentialsSecretRef.Name, Namespace: r.Options.Namespace}, foundSecret)
+		if err != nil {
+			return fmt.Errorf("failed to fetch provider secret: %w", err)
+		}
+		providerApiToken, err = getSecretContent(r.Client, provider.CredentialsSecretRef.Name, r.Options.Namespace, LLMApiTokenFileName)
+		if err != nil {
+			return fmt.Errorf("failed fetching token for provider: %s. error: %w", provider.Name, err)
+		}
+		providerCredentials += providerApiToken
+		if err = controllerutil.SetControllerReference(cr, foundSecret, r.Scheme); err != nil {
```

Comment:
I don't think we need it.
The operator doesn't manage or own user secrets.

Reply:
If we do not have this set to those secrets, the operator won't be able to look for the update events (i.e secrets credentials change) from the secrets which is required for us to update OLS app pod accordingly. We are technically not managing here but just keeping an eye on them.

Reply:
Good point. 
I thought about watcher, but it will add additional (sometimes complex) logic.

Perhaps I was just afraid of the impact on our work - I quite often create and delete OLSConfig instances and I will have to re-create secrets all the time.

@raptorsun @xrajesh any thoughts?
P.S. `lgtm` from me

Reply:
It is good to set controller reference for watching changes on the object. 
however, shall we skip setting the controller reference if it is already set? we can use this function to check if it has controller set https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#HasControllerReference

Reply:
Correct me If I am wrong, I don't this this check for controller owner reference would add any value here because user while creating those secrets will mostly not set any controller owner reference, and in case if they do end up setting it wrong our `SetContollerReference` will be there to overwrite it correctly so that updates to those secrets are listened in an expected way.